include/caffeine/Serialization/Serialize.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <string>
+// #include <mutex> // TODO remove
+#include <unordered_map>
+#include <vector>
+
+namespace caffeine {
+
+namespace serialize {
```

Comment:
You can use a compound namespace declaration to avoid having to indent what's inside.
```suggestion
namespace caffeine::serialize {
```

Reply:
I forgot I could do that, thanks